MonoidalCategories/gap/SymmetricCoclosedMonoidalCategoriesDerivedMethods.gi

```diff
@@ -919,3 +919,27 @@ AddDerivationToCAP( MonoidalPostCoComposeMorphismWithGivenObjects,
     
 end : CategoryFilter := IsSymmetricCoclosedMonoidalCategory and IsStrictMonoidalCategory,
       Description := "MonoidalPostCoComposeMorphismWithGivenObjects using coclosed evaluation, and cohom tensor adjunction" );
+
+##
+AddDerivationToCAP( IsomorphismFromInternalCoHomToCoDual,
+        
+  function( cat, a )
+    
+    return PreCompose( cat,
+                   MorphismFromInternalCoHomToTensorProduct( cat, TensorUnit( cat ), a ),
+                   RightUnitorWithGivenTensorProduct( cat, a ) );
+    
+end : CategoryFilter := IsSymmetricCoclosedMonoidalCategory,
+      Description := "IsomorphismFromInternalCoHomToCoDual using MorphismFromInternalCoHomToTensorProduct" );
+
+##
+AddDerivationToCAP( IsomorphismFromInternalCoHomToCoDualWithGivenObjects,
+        
+  function( cat, source, a, range )
+    
+    return PreCompose( cat,
+                   MorphismFromInternalCoHomToTensorProductWithGivenObjects( cat, source, TensorUnit( cat ), a, range ),
+                   RightUnitorWithGivenTensorProduct( cat, a, range ) );
```

Comment:
`RightUnitorWithGivenTensorProduct` expects the third argument to be `a_v x 1` but we have `range = a_v`.

Reply:
Right.